iroha_client_cli/src/main.rs

```diff
@@ -306,3 +313,117 @@ App::new(GET)
         }
     }
 }
+
+mod maintenance {
+    use super::*;
+    use async_std::prelude::*;
+    use clap::ArgMatches;
+    use futures::executor;
+    use iroha_client::{client::Client, config::Configuration};
+
+    const CONNECT: &str = "connect";
+    const ENTITY_TYPE: &str = "entity";
+    const EVENT_TYPE: &str = "event";
+
+    pub fn build_app<'a, 'b>() -> App<'a, 'b> {
+        App::new(MAINTENANCE)
+            .about("Use this command to use maintenance functionality.")
+            .subcommand(
+                App::new(CONNECT)
+                    .about("Use this command to connect to the peer and start consuming events.")
+                    .arg(
+                        Arg::with_name(ENTITY_TYPE)
+                            .long(ENTITY_TYPE)
+                            .value_name(ENTITY_TYPE)
+                            .help("Type of entity to consume events about.")
+                            .takes_value(true)
+                            .required(true),
+                    )
+                    .arg(
+                        Arg::with_name(EVENT_TYPE)
+                            .long(EVENT_TYPE)
+                            .value_name(EVENT_TYPE)
+                            .help("Type of event to consume.")
+                            .takes_value(true)
+                            .required(true),
+                    ),
+            )
+    }
+
+    pub fn process(matches: &ArgMatches<'_>) {
+        if let Some(ref matches) = matches.subcommand_matches(CONNECT) {
+            if let Some(entity_type) = matches.value_of(ENTITY_TYPE) {
+                println!("Connecting to consume events for: {}", entity_type);
+                if let Some(event_type) = matches.value_of(EVENT_TYPE) {
+                    println!("Connecting to consume events: {}", event_type);
+                    connect(entity_type, event_type);
+                }
+            }
+        }
+    }
+
+    fn connect(entity_type: &str, _event_type: &str) {
+        let mut iroha_client = Client::with_maintenance(
+            &Configuration::from_path("config.json").expect("Failed to load configuration."),
```

Comment:
Configuration is taken here from `config.json`, but I think it can be also specified through `--config` option. It seems here it should be taken from constant default or from param instead of a hardcoded value.

Reply:
Yes, it's an old approach with hardcoded values for CLI, put this into this issue https://jira.hyperledger.org/browse/IR-854